portfolio.py

```diff
@@ -2,13 +2,39 @@
 Allocation of the asset.
 """
 
+import numpy as np
+from thousandaire.constants import TRADING_INSTRUMENTS
+
 class Portfolio(dict):
     """
     Set the portfolio
     """
-    def __init__(self, date):
+    def __init__(self, date, target=None, np_array=None):
         dict.__init__(self)
         self.date = date
+        if target is not None:
+            if isinstance(np_array, np.ndarray):
+                self.decode_nparray(target, np_array)
+            else:
+                raise TypeError('%s type cannot be decoded.' %(type(np_array)))
```

Comment:
Can you make the message more specific? Such as `The input is not a numpy array.`